lib/db2z.js

```diff
@@ -26,32 +28,85 @@ exports.initialize = function(ds, cb) {
 };
 
 function DB2Z(settings) {
-  debug('DB2Z constructor settings: %j', settings);
   IBMDB.call(this, 'db2z', settings);
 
-  if (this.dsn) {
-    this.connStr = this.dsn;
+  // This is less than ideal, better idea would be
+  // to extend the propagation of the filter object
+  // to executeSQL or pass the options obj around
+  this.limitRE = /LIMIT (\d+)/;
+  this.offsetRE = /OFFSET (\d+)/;
+};
+
+util.inherits(DB2Z, IBMDB);
+
+DB2Z.prototype.setConnectionProperties = function(name, settings) {
+  debug('DB2Z.prototype.setConnectionProperties');
+  var self = this;
+  self.dbname = (settings.database || settings.db || 'testdb');
+  self.dsn = settings.dsn;
+  self.hostname = (settings.hostname || settings.host);
+  self.username = (settings.username || settings.user);
+  self.password = settings.password;
+  self.portnumber = settings.port;
+  self.protocol = (settings.protocol || 'TCPIP');
+
+  // Save off the connectionOptions passed in for connection pooling
+  self.connectionOptions = {};
+  self.connectionOptions.minPoolSize = parseInt(settings.minPoolSize, 10) || 0;
+  self.connectionOptions.maxPoolSize = parseInt(settings.maxPoolSize, 10) || 0;
+  self.connectionOptions.connectionTimeout =
+    parseInt(settings.connectionTimeout, 10) || 60;
+
+  var dsn = settings.dsn;
+  if (dsn) {
+    self.connStr = dsn;
+
+    var DSNObject = self.parseDSN(dsn);
+    self.schema = DSNObject.CurrentSchema || DSNObject.UID;
   } else {
     var connStrGenerate =
-      'DRIVER={db2z}' +
+      'DRIVER={' + name + '}' +
       ';DATABASE=' + this.dbname +
       ';HOSTNAME=' + this.hostname +
       ';UID=' + this.username +
       ';PWD=' + this.password +
       ';PORT=' + this.portnumber +
       ';PROTOCOL=' + this.protocol;
+    self.connStr = connStrGenerate;
 
-    this.connStr = connStrGenerate;
+    self.schema = this.username;
```

Comment:
if we want `self.schema` to be all upper case, perhaps make this `self.schema = this.username.toUpperCase()`